community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -123,24 +191,20 @@ protected void pageFault(
             // The grabFreePage method might throw.
             page = pagedFile.grabFreePage( faultEvent );
 
-            // We got a free page, and we know that we have race-free access to it.
-            // Well, it's not entirely race free, because other paged files might have
-            // it in their translation tables, and try to pin it.
-            // However, they will all fail because when they try to pin, the page will
-            // either be 1) free, 2) bound to our file, or 3) the page is write locked.
+            // We got a free page, and we know that we have race-free access to it. Well, it's not entirely race
+            // free, because other paged files might have it in their translation tables (or rather, their reads of
+            // their translation tables might race with eviction) and try to pin it.
+            // However, they will all fail because when they try to pin, the page will either be 1) free, 2) bound to
+            // our file, or 3) the page is write locked.
             stamp = page.writeLock();
-            translationTable.put( filePageId, page );
         }
         catch ( Throwable throwable )
         {
+            putObjectVolatile( chunk, chunkOffset, null );
```

Comment:
We strictly don't need to, because the `writeLock` call is uninterruptible, so it can't fail. I can add a comment about this fact, though.